ddl/index.go

```diff
@@ -468,6 +468,9 @@ func (w *worker) onCreateIndex(d *ddlCtx, t *meta.Meta, job *model.Job, isPK boo
 			} else {
 				indexInfo.Tp = indexOption.Tp
 			}
+			if indexOption.Locality == ast.IndexLocalityGlobal {
+				indexInfo.Global = true
```

Comment:
Need more strict check.
Global is set only when:
1. primary index or unique index
2. on a partition table
3. index columns does not contain all partition columns

Reply:
fixed.